Tests/ConvertTo-SecurityIdentifier.Tests.ps1

```diff
@@ -0,0 +1,116 @@
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+if( (Test-COperatingSystem -Windows) )
+{
+    & (Join-Path $PSScriptRoot -ChildPath 'Initialize-Test.ps1' -Resolve)
+
+    Describe 'ConvertTo-SecurityIdentifier.when passed in the current user SID as a string.' {
+        It 'should convert the current user SID to a SecurityIdentifier object.'{
+            $stringSID = ([System.Security.Principal.WindowsIdentity]::GetCurrent()).User.Value
+            $convertSID = ConvertTo-CSecurityIdentifier -SID $stringSID
+            $convertSID | Should -Not -BeNullOrEmpty
+            $convertSID | Should -eq $stringSID
+            $convertSID | Should -BeOfType [System.Security.Principal.SecurityIdentifier]
+        } 
+    }
+
+    Describe 'ConvertTo-SecurityIdentifier.when passed in the current user SID as a byte array.' {
+        It 'should convert the current user SID to a SecurityIdentifier object.' {
+            $SID = ([System.Security.Principal.WindowsIdentity]::GetCurrent()).User.Value
+            $encoding = [Text.Encoding]::Unicode
+            $byteSID = $encoding.GetBytes($SID)
+            $convertSID = ConvertTo-CSecurityIdentifier -SID $SID
+            $convertSID | Should -Not -BeNullOrEmpty
+            $convertSID | Should -eq $SID
+            $convertSID | Should -BeOfType [System.Security.Principal.SecurityIdentifier]
+        }
+    }
+
+    Describe 'ConvertTo-SecurityIdentifier.when passed in the current user SID as a Security Identifier object.' {
+        It 'should convert the current user SID to a SecurityIdentifier object.' {
+            $SISID = ConvertTo-CSecurityIdentifier -SID ([System.Security.Principal.WindowsIdentity]::GetCurrent()).User.Value

```

Comment:
Not sure what `$SISID` is supposed to stand for. I would rename this variable to `$SIDObj`